repl/scala-2.10/src/main/scala/org/apache/spark/repl/SparkILoop.scala

```diff
@@ -1008,7 +1008,7 @@ class SparkILoop(
     val jars = SparkILoop.getAddedJars
     val conf = new SparkConf()
       .setMaster(getMaster())
-      .setAppName("Spark shell")
+      .setAppName(sys.props.getOrElse("spark.app.name", "Spark shell"))
```

Comment:
@vanzin If `--name` is not specified on the spark-shell command line, the main class name (`org.apache.spark.repl.Main`) is set for `spark.app.name`, so `spark.app.name` will not become missing when executed via `spark-shell` command line:

https://github.com/apache/spark/blob/v1.4.1/core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala#L203-L204

The reason why I use `getOrElse` here is just for backward compatibility for special use cases (e.g., directly executing Main class of spark-shell instead of using spark-shell command line; in this case `spark.app.name` may not be set)